### PR TITLE
devices/tlv493d: Implement TLV493D hall effect sensor on I2C bus

### PR DESCRIPTION
**Pull request by bezineb5**
Implemented Infineon TLV493D, a 3-dimensional hall effect sensor:
https://www.infineon.com/cms/en/product/sensor/magnetic-sensors/magnetic-position-sensors/3d-magnetics/tlv493d-a1b6/
This is also available through Adafruit: https://www.adafruit.com/product/4366

I tried to be comprehensive in the implementation, trying to cover use cases ranging from ultra-low-power to high frequency with low precision required.

I've not included in this PR a higher-level filter to measure an angle (typical application for this sensor, see https://www.infineon.com/dgdl/Infineon-3D_Magnetic_Sensor_for_Angle_Measurements-AN-v01_10-AN-v01_01-EN.pdf?fileId=5546d46265f064ff01665a3d22f055e3) - I didn't include it as I think it's beyond the scope of periph.io, and it's full of floats. Is there any interest in such thing?

Also, I'll try to add that in the documentation (as well as the long overdue ADS1x15... sorry for that)

Thanks!
Benjamin

### CODE REVIEW

**Comment by maruel** [comment]
gohci

File: experimental/cmd/tlv493d/main.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
```
Comment (maruel) [inline experimental/cmd/tlv493d/main.go:1]: 2020

File: experimental/devices/tlv493d/doc.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2020 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package tlv493d implements interfacing code to the Infineon TLV493D haff effect sensor.
+//
+// Features of the device:
+// - 3-dimensional hall effect sensor, measures up to +/-130 mT magnetic flux.
```
Comment (maruel) [inline experimental/devices/tlv493d/doc.go:8]: It won't show up nicely in godoc sadly. You'll probably want to do indent so it shows as a code block.

File: experimental/devices/tlv493d/doc.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// - temperature sensor
+// - i2c interface
+// - 12-bit resolution
+// - low power consumption
+//
+// Features of the driver:
+// - Implemented all options of the device
```
Comment (maruel) [inline experimental/devices/tlv493d/doc.go:15]: same

File: experimental/devices/tlv493d/example_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
```
Comment (maruel) [inline experimental/devices/tlv493d/example_test.go:1]: 2020

File: experimental/devices/tlv493d/tlv493d.go [hunk 5]
```diff
@@ -0,0 +1,473 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
```
Comment (maruel) [inline experimental/devices/tlv493d/tlv493d.go:1]: 2020

File: experimental/devices/tlv493d/tlv493d.go [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+	bitTemperatureMeasurement = 7
+	bitParityTest             = 5
+
+	magneticFluxScaling   = 98 * physic.MicroTesla
+	temperatureScaling    = 1100 * physic.MilliCelsius
+	temperatureScalingDiv = 10
+	referenceTemperature  = 25*physic.Celsius + physic.ZeroCelsius
+)
+
+// Mode reprents the vairous power modes described in the documentation
```
Comment (maruel) [inline experimental/devices/tlv493d/tlv493d.go:67]: various

File: experimental/devices/tlv493d/tlv493d_test.go [hunk 7]
```diff
@@ -0,0 +1,345 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
```
Comment (maruel) [inline experimental/devices/tlv493d/tlv493d_test.go:1]: 2020

**Review by maruel — APPROVED** [review]
Thanks a lot! Looks great, only minor nits, nothing really blocking.

I just want to update the static file as a separate change first, then this can be committed.

**Comment by bezineb5** [comment]
Thanks for the comments! I fixed the typos.

**Comment by maruel** [comment]
Thanks for the PR and apologies for the delay.

**Pull request merged**